app/move/steps/create.js

```diff
@@ -16,32 +15,48 @@ module.exports = {
     resetJourney: true,
     skip: true,
     checkJourney: false,
-    next: 'pnc-search',
+    next: [
+      {
+        fn: req => {
+          return req.session.currentLocation.location_type === 'prison'
+        },
+        next: 'nomis-search',
+      },
+      'pnc-search',
+    ],
+  },
+  '/nomis-search': {
+    entryPoint: true,
+    template: 'move/views/create/nomis-search',
+    pageTitle: 'moves::steps.nomis_offender_no_search_term.heading',
+    buttonText: 'actions::continue',
+    next: 'person-search-results',
+    fields: ['nomis_offender_no_search_term'],
   },
   '/pnc-search': {
-    checkJourney: false,
-    backLink: false,
-    controller: Base,
-    method: 'get',
-    action: '/move/new/pnc-search-results',
+    entryPoint: true,
     template: 'move/views/create/pnc-search',
     pageTitle: 'moves::steps.police_national_computer_search_term.heading',
     buttonText: 'actions::continue',
-    next: 'pnc-search-results',
+    next: 'person-search-results',
     fields: ['police_national_computer_search_term'],
   },
-  '/pnc-search-results': {
-    checkJourney: false,
```

Comment:
I believe checkJourney was disabled here so the previous conditions aren't validated - with this removed you can get in a redirect loop between personal details and search results.